app/services/activity/grouped_activities_fetcher.rb

```diff
@@ -0,0 +1,41 @@
+class Activity
+  class GroupedActivitiesFetcher
+    def initialize(user:, organisation:, scope: :all)
+      @organisation = organisation
+      @scope = scope
+      @activities = ActivityPolicy::Scope.new(user, Activity).resolve
+    end
+
+    def call
+      grouped_programmes.each_with_object({}) do |(fund, programmes), funds|
```

Comment:
This is super clever, it took a while for me to unpick what was going on, but that is my ignorance! 👍 

Question: this kind of nested looping over the hierarchy is probably common, is there a way to split the nesting in to method calls to make it approachable?

Reply:
It's a bit tricky I think - we're acting on an ActiveRecord relation, and we're having to do something with each thing when we loop through the relation. It might be something we can improve on later, but it's making my head hurt! OK to let this go in and improve on it later?